sdk/identity/Azure.Identity/api/Azure.Identity.netstandard2.0.cs

```diff
@@ -106,6 +106,7 @@ public partial class DefaultAzureCredential : Azure.Core.TokenCredential
     {
         public DefaultAzureCredential(Azure.Identity.DefaultAzureCredentialOptions options) { }
         public DefaultAzureCredential(bool includeInteractiveCredentials = false) { }
+        public System.Type CredentialType { get { throw null; } }
```

Comment:
SelectedCredentialType? Any reason not to expose credential itself?

Reply:
What would you do with the full credential that you couldn't do with the instance of DefaultAzureCredential itself?

Reply:
Call `.GetType()` on it :)

Reply:
True, I just wonder if it would introduce confusion - should I use it to call `GetToken`? If I do, is that ever different than calling `GetToken` on the instance of `DefaultAzureCredential`?

Reply:
Can I use the same `DefaultAzureCredential` instance for multiple clients?  And could it result in each client using different credentials?  Could I use this API reliably if those were on different threads?  If so, this behaves kind of like `GetLastError()` which is a very challenging API for many customers.

I think exposing this as a property with this name implies too much confidence in the result.  What if we had `TokenCredential GetMostRecentCredential()` or something uglier like that to help avoid confusion about how it's meant to be used?

Reply:
I would also like to know what exact runtime scenario we are enabling with this propery.

Reply:
The credential is acquired for the first time under `AsyncLockWithValue` so there shouldn't be any threading issues there. I don't think it is possible to have an instance of DAC that serves up more than one credential type.

Reply:
> I would also like to know what exact runtime scenario we are enabling with this property. Reading the original issue it seems to be diagnostics related.

I'm not aware of any. @jongio do you have any perspective to add here?

Reply:
When debugging auth issues such as permissions, etc. It is helpful to know which credential type is being used and what client id was selected. 

For example, if you are getting permission / rbac errors and you are trying to track down if the app is using Azure CLI so you can give your CLI id perms or if it is using something else, like VS Code or Managed Identity.

Reply:
Can we add a DebuggerVisualizer instead?

Reply:
Could we also (I like the DebuggerVisualizer idea) improve logging to make it more obvious which was selected - at least a specific event ID people could filter with instruction without adding all the reasons other ones failed. Another benefit here is that you don't need to change or debug into code to get an easy view on which credential type was used.

Reply:
Decided to just log a more specific event for now to get this merged, and we can consider a visualizer later.